lib/core/metadata.cpp

```diff
@@ -0,0 +1,59 @@
+#include "metadata.hpp"
+
+#include "errors.h" // for CHECK_ERROR_F, CHECK_MEM_F
+
+#include <assert.h> // for assert
```

Comment:
Maybe use `cassert`, `cstdlib`, and `cstring` here instead?

Is `cstring` even necessary?

Reply:
fixed!